C22_Physics/SteveManager.h

```diff
@@ -0,0 +1,67 @@
+/*----------------------------------------------
+Programmer: Ruben Young (ray6190@rit.edu)
+Date: 2019/11
+----------------------------------------------*/
+#ifndef STEVEMANAGER_H
+#define STEVEMANAGER_H
+
+#include "Definitions.h"
+#include "LaneGrid.h"
+
+
+namespace Simplex{
+class SteveManager
+{
+    uint m_steveCount = 0;
+    LaneGrid* m_pLaneGrid = nullptr;
+
+public:
+    // Default Constructor and Destructor
+    SteveManager();
+    ~SteveManager();
+
+    void Init();
+
+    // This method handles:
+    // - Checking for dead Steves 
+    // - Moving Steves forward
+    // - Spawning new Steves (if necessary)
+    void Update();
+
+    #pragma region Singleton-specific method definitions + implementation
+    // Allows external agents access to the SteveManager singleton through static pointer
+    static SteveManager * GetInstance()
+    {
+        if (m_pInstance == nullptr)
+        {
+            m_pInstance = new SteveManager();
+        }
+        return m_pInstance;
+    }
+
+    static void ReleaseInstance()
+    {
+        if (m_pInstance != nullptr)
+        {
+            delete m_pInstance;
+            m_pInstance = nullptr;
+        }
+    }
+    #pragma endregion
+
+private: // Methods
+
+    // Steve-Spawning Methods + Overrides
+    void SpawnSteve(uint a_laneIndex = 0);  // Random direction vector
+    void SpawnSteve(vector3 a_direction, uint a_laneIndex = 0); // Specific direction vector
+
+    void UpdateSteveCollisions();      // Bounce steves off each other
```

Comment:
Take a look at the `MyEntityManager` class. If you use that existing class when creating steves it will automatically do physics solving for you in its `update` function as long as the steves are in the right dimension (LaneGrid determines dimension as long as you pass the steve entity into the `AddToLane` function). That's why I wouldn't even bother with having this function and just make sure that the AIs have "applies force" in the direction of the player.